src/components/TransportTime.js

```diff
@@ -29,12 +31,52 @@ class TransportTime extends React.Component {
         compact: false,
     };
 
+    constructor() {
+        super();
+        this.state = {};
+    }
+
     // flow:disable not supported yet
     get compactClass(): string {
         return this.props.compact ? "compact" : "";
     }
 
+    loadTime(): void {
+        const destPoint = this.props.location.point;
+        const coords = storage.getJSON("coordinates");
+        let location = storage.getItem("location");
+
+        if (coords && coords.latitude && coords.longitude) {
+            location = `${coords.latitude},${coords.longitude}`;
+        }
+        // FIXME: check google.maps.DirectionsTravelMode.TRANSIT
+        // once we expand beyond VIC (or transit is implemented in VIC).
+
+        const options = {
+            origin: `${location}`,
+            destination: `${destPoint.lat},${destPoint.lon}`,
+            travelMode: google.maps.DirectionsTravelMode.WALKING,
+            unitSystem: google.maps.UnitSystem.METRIC,
+            language: "en-AU",
+        };
+
+        let directionsService = new google.maps.DirectionsService();
+
+        directionsService.route(options, (response, status) => {
+            if (status == google.maps.DirectionsStatus.OK) {
+                this.setState({time: response.routes[0].legs.map(
+                    (leg) => leg.duration.text
+                ).reduce(
+                    (memo, text) => `${memo} then ${text}`
+                )});
```

Comment:
Probably doesn't even need humanising? Also if we get the cost field, should we include that?